File: src/SistemaComercial.Domain/Entities/Caja.cs
using System.Text.Json.Serialization;
using SistemaComercial.Domain.Entities;
public class Caja
{
    public short IdCaja { get; set; }
    public string NumeroCaja { get; set; } = string.Empty;
    public short IdUsuario { get; set; }
   
    [JsonIgnore]
    public Usuario? Usuario { get; set; }
    public DateTime FechaApertura { get; set; }
    public DateTime? FechaCierre { get; set; }
    public decimal SaldoInicial { get; set; }
    public decimal? SaldoFinal { get; set; }
    public decimal? SaldoEsperado { get; set; }

    public decimal? Diferencia { get; set; }
    public string? Observaciones { get; set; }
    public string Estado { get; set; } = "Abierta";

    // Navegación
    [JsonIgnore]
    public ICollection<MovimientoCaja> MovimientosCaja { get; set; } = [];
    
    [JsonIgnore]
    public ICollection<Venta> Ventas { get; set; } = [];
}

File: frontend/src/api/caja.ts
import { api } from "./axios";

export interface Caja {

    idCaja: number;

    numeroCaja: string;

    idUsuario: number;

    fechaApertura: string;

    fechaCierre: string | null;

    saldoInicial: number;

    saldoFinal: number;

    saldoEsperado: number;

    diferencia: number;

    observaciones: string;

    estado: string; // "Abierta" | "Cerrada"

}

export interface AbrirCajaInput {
    saldoInicial: number;
    observaciones: string;
}

// Ojo: tu Swagger muestra el mismo body (saldoInicial + observaciones)
// para cerrar. Lo dejo tal cual lo pasaste; si en la práctica el cierre
// necesita el saldo final contado, avisame y lo ajustamos.
export interface CerrarCajaInput {
    saldoInicial: number;
    observaciones: string;
}

export async function obtenerCajas() {

    const response = await api.get<Caja[]>(
        "/Caja",
    );

    return response.data;

}

export async function abrirCaja(data: AbrirCajaInput) {

    const response = await api.post<Caja>(
        "/Caja/abrir",
        data,
    );

    return response.data;

}

export async function cerrarCaja(
    { id, data }: { id: number; data: CerrarCajaInput },
) {

    const response = await api.put<Caja>(
        `/Caja/cerrar/${id}`,
        data,
    );

    return response.data;

}